j5/backends/hardware/sr/v4/motor_board.py

```diff
@@ -97,6 +100,11 @@ def __del__(self) -> None:
     @handle_serial_error
     def send_command(self, command: int, data: Optional[int] = None) -> None:
         """Send a serial command to the board."""
+        with self._lock:
```

Comment:
I think the lock only needs to be held for the call to `self._serial.write(bytes(message))`, but it's not a big deal.

Reply:
c09b737

Reply:
This was deliberate to try and reduce duplicate code.

If you think that it is important that the lock is only held for the IO operation, I can change it, but context handlers have a side effect of creating duplicate code.

Although we could have a second `null` lock context handler that doesn't do this I suppose..

Reply:
I'd be interested in some more opinions on this.

@sedders123 @tpoomlmly

Reply:
I have done 1c6ea21 as an example of my suggestion.

Reply:
Hmm, so looking at the current code again now it's obvious that's how `_send_command_no_lock` is being used (I was aware of this when I originally reviewed the code too so I have no idea what was going through my head when I left that comment).

1c6ea212b37fbf23b51c1eea0daf267452e945cc is a neat solution, but I think it's less straightforward than what we had originally - I think the original solution with `_send_command_no_lock` is easier to understand.

Sorry for the wild goose chase here.

Reply:
9ce14b9